avocado/utils/network/interfaces.py

```diff
@@ -571,3 +572,22 @@ def restore_slave_cfg_file(self):
                     os.remove(slave_config)
             except Exception as ex:
                 raise NWException(f"Could not restore the config file {ex}")
+
+    def check_packet_loss(self, peer_ip, count, option=None, flood=False):
+        """
+        Function checks for packet loss during ping.
+        Return False for 0% packet loss and True if packet loss occurs.
+        """
+        cmd = f'ping -I {self.name} {peer_ip} -c {count}'
+        if flood is True:
+            cmd = f'{cmd} -f'
+        elif option is not None:
+            cmd = f'{cmd} {option}'
+        try:
+            output = process.system_output(cmd, ignore_status=True,
```

Comment:
The entire `network/*` modules is being used for `LocalHost` and `RemoteHost`. `process.system_output()` should be avoided (in fact there is not call for it, except on `common.py`).  `common.run_command()` should be used instead to support both scenarios.

Reply:
Also, can't [ping_check()](https://avocado-framework.readthedocs.io/en/96.0/api/utils/avocado.utils.network.html#avocado.utils.network.interfaces.NetworkInterface.ping_check) be used for the same purposes here?

Reply:
@beraldoleal Sure will try using run.command()  in-place of process.system_output().

@clebergnu  Ping_check()  doesn't verify the packet loss is 0% or not. Just to make sure there is no packet loss this function was added. This can be called with wait_for  function which can be having some time to wait until the condition is satisfied based on the return value of above util function. like below

if not wait.wait_for(lambda: self.networkinterface.check_packet_loss(
                             self.peer, count=10000, option='-f'), timeout=30):
            self.cancel("Packet recieved in Ping flood is not 100 percent \
                         after waiting for 30sec")

Reply:
@beraldoleal yes right, we should be, apologies, @TasmiyaNalatwad can you please send PR with changes

Reply:
@PraveenPenguin  Sure, I am on it.